### DISCUSSION TITLE
report error about namespace std

### DISCUSSION THREAD

**Discussion by codezijie** [clangd/clangd#1180] [Q&A]
if I use namespace std, the clangd will report : 
![image](https://user-images.githubusercontent.com/29241509/172041608-21f1cf13-0102-460f-b3dc-8d7428e70d18.png)
even, some my namespace have the same problem:
![image](https://user-images.githubusercontent.com/29241509/172041657-5bcabb00-f497-4db5-b918-3004efd98179.png)

**Comment by HighCommander4** [comment]
Please provide a complete code example; these screenshots are not enough to go on.

**Comment by love1angel** [comment]
@HighCommander4 hi, following are example, i use cmake for building

``` C++
namespace {
const std::string domain{"local"};
const std::string c_capiConnectionId{"test"};
}  // namespace
```
<img width="889" alt="image" src="https://user-images.githubusercontent.com/45203165/194701977-d000c9be-ce53-4a32-9a86-e4443edb83ad.png">

**Reply by HighCommander4** [reply]
The example is missing `#include <string>`.

**Reply by love1angel** [reply]
it has include string, only part of the code is captured, in my project all the std cannot be resolved correctly

**Reply by HighCommander4** [reply]
I can't really make useful suggestions based on a partial code example. Please show a complete code example. [Clangd logs](https://clangd.llvm.org/troubleshooting#gathering-logs) would be helpful as well.

**Reply by love1angel** [reply]
hi, @HighCommander4 @codezijie, i add "clangd.arguments" and specify the compiler path, and now can resolve normally
```
{
    "clangd.arguments": [
        "-header-insertion=never",
        "--query-driver=/opt/heli23/sysroots/x86_64-pokysdk-linux/usr/bin/aarch64-poky-linux/aarch64-poky-linux-g++",
        "--log=verbose",
        "--pretty"
    ]
}
```
